dali/pipeline/pipeline.h

```diff
@@ -385,6 +385,16 @@ class DLL_PUBLIC Pipeline {
    */
   DLL_PUBLIC inline int device_id() const { return device_id_; }
 
+    /**
+   * @brief Returns number of outputs.
+   */
+  DLL_PUBLIC int num_outputs() const;
+
+  /**
+   * @brief Returns string describing device backing output specified by given id.
+   */
+  DLL_PUBLIC std::string output_device(int id) const;
```

Comment:
```suggestion
  DLL_PUBLIC const std::string &output_device(int id) const;
```
This will avoid unnecessary copy when using it like `pipe.output_device(id) == "gpu"`.

Reply:
Done